securedrop_client/gui/conversation/print/confirmation_dialog.py

```diff
@@ -0,0 +1,91 @@
+from gettext import gettext as _
+
+from PyQt5.QtCore import QSize, pyqtSlot
+
+from securedrop_client.gui.base import ModalDialog, SecureQLabel
+
+from .printer import Printer
+
+
+class ConfirmationDialog(ModalDialog):
+
+    FILENAME_WIDTH_PX = 260
+
+    def __init__(self, printer: Printer, file_name: str) -> None:
+        super().__init__()
+
+        self._printer = printer
+        self._printer.status_changed.connect(self._on_printer_status_changed)
+
+        self.continue_button.setText("PRINT")
+        self.continue_button.clicked.connect(
+            self.accept
+        )  # FIXME The ModalDialog is more complex than needed to do this.
+
+        file_name = SecureQLabel(
+            file_name, wordwrap=False, max_length=self.FILENAME_WIDTH_PX
+        ).text()  # FIXME This seems like a heavy way to sanitize a string.
+        header = _("Print:<br />" '<span style="font-weight:normal">{}</span>').format(file_name)
+        body = _(
+            "<h2>Managing printout risks</h2>"
+            "<b>QR codes and web addresses</b>"
+            "<br />"
+            "Never type in and open web addresses or scan QR codes contained in printed "
+            "documents without taking security precautions. If you are unsure how to "
+            "manage this risk, please contact your administrator."
+            "<br /><br />"
+            "<b>Printer dots</b>"
+            "<br />"
+            "Any part of a printed page may contain identifying information "
+            "invisible to the naked eye, such as printer dots. Please carefully "
+            "consider this risk when working with or publishing scanned printouts."
+        )
+
+        self.header.setText(header)
+        self.header_icon.update_image("printer.svg", svg_size=QSize(64, 64))
+        self.body.setText(body)
+        self.adjustSize()
+
+        self._body = body
+
+        self._on_printer_status_changed()
+
+    @pyqtSlot()
+    def _on_printer_status_changed(self) -> None:
+        printer_status = self._printer.status
+        if printer_status == Printer.StatusUnknown:
+            self._on_printer_status_unknown()
+        elif printer_status == Printer.StatusReady:
+            self._on_printer_ready()
+        elif printer_status == Printer.StatusUnreachable:
+            self._on_printer_unreachable()
+        else:  # Printer.StatusBusy
+            self._on_printer_busy()
+
+    def _on_printer_status_unknown(self) -> None:
+        self.continue_button.setEnabled(False)
+        self.printer_start_requested.emit()
+
+        status = "<i>Waiting for printer status to be known...</i>"
+        self.body.setText("<br /><br />".join([self._body, status]))
+        self.adjustSize()
```

Comment:
This looks like it wants to call a `self.set_status()` method like:

```python
class ConfirmationDialog(ModalDialog):
    # [...]

    def set_status(status: str) -> None:
        # TODO: handle case of *removing* the status
        self.body.setText("<br /><br />").join([self._body, status]))
        self.adjustSize()
```

What do you think?

Reply:
Yes. There are a few things wrong with this "append a string" approach. E.g. `self.adjustSize()` doesn't have the desired effect (making the last line visible). So there are a few things to revisit. 

Intuitively, I think that showing and hiding widgets in a layout will be easier conceprually and probably solve the visibility issues without need for explicit handling. To be confirmed.

No matter how we do that auxiliary method to show/hise/select the adequate widget/copy may well make sense indeed 👍

I'm planning to look into that when writing tests for the dialogs.